src/xtea.cpp

```diff
@@ -28,48 +28,77 @@ namespace xtea {
 
 namespace {
 
-constexpr uint32_t delta = 0x9E3779B9;
+void writeBlock(uint32_t left, uint32_t right, uint8_t data[blockSize])
+{
+	data[0] = static_cast<uint8_t>(left);
+	data[1] = static_cast<uint8_t>(left >> 8u);
+	data[2] = static_cast<uint8_t>(left >> 16u);
+	data[3] = static_cast<uint8_t>(left >> 24u);
+	data[4] = static_cast<uint8_t>(right);
+	data[5] = static_cast<uint8_t>(right >> 8u);
+	data[6] = static_cast<uint8_t>(right >> 16u);
+	data[7] = static_cast<uint8_t>(right >> 24u);
+}
 
-template<typename Round>
-void apply_rounds(uint8_t* data, size_t length, Round round)
+// applies 64 rounds of xtea to a 64bit block
+void encryptBlock(uint8_t data[blockSize], const key& k)
 {
-	for (auto j = 0u; j < length; j += 8) {
-		uint32_t left = data[j+0] | data[j+1] << 8u | data[j+2] << 16u | data[j+3] << 24u,
-				right = data[j+4] | data[j+5] << 8u | data[j+6] << 16u | data[j+7] << 24u;
-
-		round(left, right);
-
-		data[j] = static_cast<uint8_t>(left);
-		data[j+1] = static_cast<uint8_t>(left >> 8u);
-		data[j+2] = static_cast<uint8_t>(left >> 16u);
-		data[j+3] = static_cast<uint8_t>(left >> 24u);
-		data[j+4] = static_cast<uint8_t>(right);
-		data[j+5] = static_cast<uint8_t>(right >> 8u);
-		data[j+6] = static_cast<uint8_t>(right >> 16u);
-		data[j+7] = static_cast<uint8_t>(right >> 24u);
+	uint32_t left = data[0] | data[1] << 8u | data[2] << 16u | data[3] << 24u,
+		right = data[4] | data[5] << 8u | data[6] << 16u | data[7] << 24u;
+	int_fast8_t i = 0;
+
+	while (i < rounds) {
```

Comment:
@ranisalt agreed, that's a leftover i missed from an early version i made with c, it just my habit to use _fast_ or _least_ instead of fixed intN_t there because _fast_ and _least_ is portable while fixed is not (by c specs, im not aware how c++ defines them)

Reply:
I think C++ just inherits the definition from C, but modern computers don't have that much difference and modern compilers know when to promote integers for performance purposes, so it's pretty much useless :P